node/narwhal/src/primary.rs

```diff
@@ -1053,3 +1053,317 @@ impl<N: Network> Primary<N> {
         self.gateway.shut_down().await;
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::MAX_EXPIRATION_TIME_IN_SECS;
+
+    use snarkos_node_narwhal_committee::MIN_STAKE;
+    use snarkos_node_narwhal_ledger_service::MockLedgerService;
+
+    use bytes::Bytes;
+    use rand::RngCore;
+
+    type CurrentNetwork = snarkvm::prelude::Testnet3;
+
+    // Returns a primary and a list of accounts in the configured committee.
+    async fn primary_without_handlers(
+        rng: &mut TestRng,
+    ) -> (Primary<CurrentNetwork>, Vec<(SocketAddr, Account<CurrentNetwork>)>) {
+        // Create a committee containing the primary's account.
+        let (accounts, committee) = {
+            const COMMITTEE_SIZE: usize = 4;
+            let mut accounts = Vec::with_capacity(COMMITTEE_SIZE);
+            let mut members = IndexMap::new();
+
+            for i in 0..COMMITTEE_SIZE {
+                let socket_addr = format!("127.0.0.1:{i}").parse().unwrap();
```

Comment:
shouldn't the port range start somewhere later? some of the low ports are reserved

Reply:
Didn't have issue funilly enough, good catch!